includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1402,6 +1435,11 @@ protected function build_response( $itemized_display_items = false, $has_shippin
 		// Default show only subtotal instead of itemization.
 		if ( ! apply_filters( 'wc_stripe_payment_request_hide_itemization', true ) || $itemized_display_items ) {
 			foreach ( WC()->cart->get_cart() as $cart_item_key => $cart_item ) {
+				// Exclude bundled items to avoid making the list too big
+				if ( function_exists( 'wc_pb_is_bundled_cart_item' ) && wc_pb_is_bundled_cart_item( $cart_item ) ) {
+					continue;
+				}
```

Comment:
I've decided to do this because when the bundle has two or more items the payment dialog will show only a few items and will hide the other items as well as Tax and Shipping amounts. I think it's more important to show Tax and Shipping than the full list of items in the bundle.

<img width="567" alt="Screen Shot 2021-06-17 at 11 42 42 PM" src="https://user-images.githubusercontent.com/1025173/122507373-b65d3200-cfc5-11eb-9165-4eb4562b3d39.png">

<img width="527" alt="Screen Shot 2021-06-17 at 11 42 49 PM" src="https://user-images.githubusercontent.com/1025173/122507383-bb21e600-cfc5-11eb-9eec-42e996797721.png">

Reply:
Makes sense 👍

Reply:
Although this is also a special check for a different extension, I think this one is different because it belongs to our codebase.

The payment request labels is totally our responsibility.

Reply:
I'm still feeling on the fence about this. I've read some comments related to this from @reykjalin here p1624282593131400/1623781574.092400-slack-C01B82V67DL 

Some input from @reykjalin on this would be nice. Personally, I think it's better not to show all child items otherwise we will end up hiding tax and shipping amounts.

Reply:
To be clear about my previous comment to this thread, I think it's fine to have a special check for payment request labels, since they're all our responsibility, in contrast to the `needs_shipping` part.

Although it's better for UX and hides the `$0` "sub-labels", I'm also unsure about hiding some of the options.

Reply:
> I think it's more important to show Tax and Shipping than the full list of items in the bundle.

Bundles may contain items that are "Priced Individually" -- the safest way to approach the problem here is to display a line for every bundled item. Some merchants may ask for more flexibility here, or may want this specific view to respect some of plugin's cart display options, such as:

- [Item Grouping](https://docs.woocommerce.com/document/bundles/bundles-configuration/#grouping-options) which basically "hides" the parent item;
- [Visibility](https://docs.woocommerce.com/document/bundles/bundles-configuration/#advanced-settings) (used to keep certain bundled items "hidden" from customers).

Reply:
Hmm, I see your point here @asumaran about how it would be confusing if the Subtotal, Tax, and Shipping lines aren't present 🤔 

At the same time, the dialog seems to have the cut-off at 2 items? Wouldn't one of the Subtotal, Tax, or Shipping disappear anyway? Maybe we can just respect the `wc_stripe_payment_request_hide_itemization` filter here so the merchant can configure this the way they want?

I honestly think it might be best to bring in @LevinMedia to get a designers 👀 on the UX part of the experience here.

@asumaran can you provide screenshots for comparing the "complete" version versus the "hidden" version? I think it'll be easier for @LevinMedia to evaluate which approach might be better if we can do that 🙂